orangecontrib/spectroscopy/utils/__init__.py

```diff
@@ -111,7 +111,7 @@ def get_ndim_hyperspec(data, attrs):
     new_shape = tuple([lsa[2] for lsa in ls]) + (data.X.shape[1],)
     hyperspec = np.ones(new_shape) * np.nan
 
-    hyperspec[indices] = data.X
+    hyperspec[tuple(indices)] = data.X
```

Comment:
I'd say this is the most important fix in this PR. But it can be be dangerous. Calling `tuple()` on something copies it and also makes the structure slower for numpy to work with it later.

While your changes fix a potential future bug, I'd prefer fixing it in other way: to have indices saved directly in a proper numpy structure. This probably includes changing the output of axes_to_ndim_linspace, maybe just doing some kind of stacking on indices there.

Reply:
Well, let's try to avoid filling the memory and go with what you suggest.